pallets/parachain-staking/src/lib.rs

```diff
@@ -1502,6 +1480,50 @@ pub mod pallet {
 			<SelectedCandidates<T>>::get().binary_search(acc).is_ok()
 		}
 
+		pub fn join_candidates_inner(
```

Comment:
Maybe I'm missing something but doesn't the title state that we need to remove this constraint from `moonbeam-orbiters::orbiter_register()` call in the end. I think we're missing some code in that pallet as well.

Reply:
@rimbi can you update the titel and  description of the PR to match the new design decision?

Reply:
You are right. The title was for the first solution and did not reflect the purpose of the current solution. Changed. 👍